app/models/authentication.rb

```diff
@@ -0,0 +1,20 @@
+class Authentication
+
+  def initialize(parser)
+    @parser = parser
+  end
+
+  def current_user(request)
+    @request = request
+    if @request.headers['Authorization']
```

Comment:
to avoid the nesting inside the if, can you do an early return when there's no header instead?